### PR TITLE
feat: add technical architecture overview

### LINKED ISSUE #1030: Add an overview of the architecture of the site.
Expand beyond what is summarized in the [Folder Structure](https://github.com/Qiskit/qiskit.org/#-folder-structure) section of the README.md to explain how the site is architected including:
- Components.
- Integrations with external services.
- Markdown content rendering (old and new styles).
- Implementation of Carbon Design System

### PR DESCRIPTION
**Pull request by eddybrando**
Expand technical architecture overview section in the **README.md** file as suggested in #1030.

The newly created **ARCHITECTURE.md** aims to serve as a simple but coherent and complete overview of how the website is architected, without going into too much detail.

It should be seen as a general guide of our current practices and a good onboarding document for project newcomers.

Some updates have been also made to the **README.md** file to reflect the current directory structure and fix minor typos.

---

Closes #1030
Replaces #1225

### CODE REVIEW

File: ARCHITECTURE.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+## Vue.js components
+
+We write our components as [single file components](https://vuejs.org/v2/guide/single-file-components.html).
+
+The **script section** is written in [class-style syntax](https://class-component.vuejs.org/) with TypeScript and uses decorators provided by [**vue-property-decorator**](https://github.com/kaorun343/vue-property-decorator).
+
+The **style section** is written in SCSS.
+
+Here is an example of a component called `GreetingMessage.vue`:
```
Comment (eddybrando) [inline ARCHITECTURE.md:23]: From https://github.com/Qiskit/qiskit.org/pull/1225#discussion_r560228814 by @delapuente:

> Would be convenient to provide an example based on a real one.

My goal with this code snippet is to give a very quick glance at the use of TypeScript and SCSS in our Vue components. I believe that using a real example may add more elements and divert to attention to other more complex parts of a component.

Nevertheless I see value in displaying a code snipped with an example of a simplified version of a real project component.

File: ARCHITECTURE.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+## Markdown content
+
+We use Markdown files to add content and have two different approaches for this:
+
+- content (old)
+- new-content (new)
+
+These approaches currently coexist but we aim to replace the former with the latter.
+
+### content (old)
```
Comment (eddybrando) [inline ARCHITECTURE.md:95]: From https://github.com/Qiskit/qiskit.org/pull/1225/files#r560230565 by @delapuente:

> Would be convenient to mention the front-matter & markdown plugin we are using.

**Review by delapuente — APPROVED** [review]
Let's unblock this. There will be more opportunities to improve it!

**Comment by delapuente** [comment]
@y4izus please take a look, add your comments and let's merge this.

@eddybrando nice done, please update whatever you consider it should be updated and prepare the branch for merging.

File: README.md [hunk 3]
```diff
@@ -80,13 +80,13 @@ Qiskit.org is a pre-rendering SPA using [Nuxt.js](https://nuxtjs.org/).
 
 A **pre-rendering SPA** is a single page application that generates a static markup (HTML) at build time. The user, when entering the web, receives HTML (as if it were a static web) but in the meantime, JS files belonging to the SPA are loaded “hydrating” the web until it's completely dynamic.
 
-**[Nuxt.js](https://nuxtjs.org/)** is the biggest framework on top of **[Vue.js](https://vuejs.org/)** to generate *universal* SPAs. Universal or "isomorphic" apps can be pre-rendering or SSR. Since so far we don't need server functions, our website it's just pre-rendering.
+**[Nuxt.js](https://nuxtjs.org/)** is the biggest framework on top of **[Vue.js](https://vuejs.org/)** to generate *universal* SPAs. Universal or "isomorphic" apps can be pre-rendering or SSR. Since so far we don't need server functions, our website is just pre-rendering.
 
 We create and run unit tests using [Jest](https://jestjs.io/), ensure avoiding syntax errors using [ESLint](https://eslint.org/) and automate all these previous tools and deployment using [Travis](https://travis-ci.org/).
```
Comment (y4izus) [inline README.md:85]: Right now we are using Github actions instead of travis

**Review by y4izus — CHANGES_REQUESTED** [review]
Great job Eddy! just one comment and we can merge :D

**Review by delapuente — APPROVED** [review]

**Pull request merged**